### ISSUE TITLE
Add "suggest" to SearchResponse / Error during CompletionSuggester (8.x Client)

### ISSUE THREAD

**Issue by svenhartmann** [elastic/elasticsearch-net#7390]
1. It looks like the `SearchResponse` is missing serialization for a suggestion response yet, e.g. https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters.html#querying

2. Usage of the `Prefix` property at  `CompletionSuggester` level leads to an error -> `[9:9] [completion] unknown field [prefix]`

Not working example:
```
new CompletionSuggester
{
    Field = "myfield", 
    Size = 20, 
    Prefix = "Act",
    SkipDuplicates = true,
}
```

Working example:
```
var suggester = FieldSuggester.Completion(new CompletionSuggester
{
    Field = "myfield", 
    Size = 20, 
    SkipDuplicates = true,
});
suggester.Prefix = "Act";
```

**Comment by stevejgordon** [comment]
Thanks for raising this @svenhartmann. The suggesters property had some issues with code-gen and is on a list for me to revisit. I'll keep this issue open to track that progress. For the second item, I'll need to investigate. That might be an issue with the spec.

**Comment by stevejgordon** [comment]
@svenhartmann For the second item, the second code snippet is the correct way to configure the suggester. We model this in the spec as a specialised container concept, with `FieldSuggester` being able to contain one of the three possible suggester variants (completion, phrase and term). There is an issue with our spec as prefix (and regex) shouldn't appear on the `CompletionSuggester` variant. I'm going to split this into two issues so we can track and resolve each independently.

**Comment by svenhartmann** [comment]
@ste

**Comment by svenhartmann** [comment]
@stevejgordon Would be nice to have this feature and the range queries within the next release. Is it feasible?

**Comment by svenhartmann** [comment]
@stevejgordon Any news here?

**Comment by stevejgordon** [comment]
@svenhartmann I looked at this for 8.1.0, but we still need to do some work in code-gen to handle the spec and make the code usable. I don't have an ETA for when we will pick this up but agree it's a gap we need to fill.

**Comment by svenhartmann** [comment]
@stevejgordon Seems that its not in the 8.1.1 release. I started migration of a webshop search of a customer in march and the only features that breaks the upgrade is the missing suggest response.

**Comment by stevejgordon** [comment]
@svenhartmann I didn't get a chance to focus on this for the latest release, so unfortunately, it is still open. I have moved on from maintaining this library day-to-day which has slowed progress, and we have a new engineer starting today to take ownership. We can't provide any timescales for this item at the moment, but it will be picked up when possible.

**Comment by asal-hesehus** [comment]
I am also migrating to the new client and are also blocked by this missing feature.

**Comment by TheFireCookie** [comment]
Hello, we are also trying to migrate to Elastic.Clients.Elasticsearch 8.1.x (to upgrade also from ES7 to ES8) and we also have this issue with the suggestions that is blocking our upgrade. Could we provide you some help ? If you give me some guidance, I can take a look to fix that issue?

**Comment by Robbware** [comment]
Hello, I've also ran into this roadblock for an upgrade on our current project and would appreciate if this could be added!

**Comment by timschwallie** [comment]
It looks possible to use 
string response = resultQuery.ApiCallDetails.ResponseBodyInBytes.ToUtf8String();
to grab the response in json format.
From there it looks like the "suggest" isn't set up right for an easy conversion to c# objects. (using json2csharp.com). Basically, it sets up each suggestor as parameters rather than an array. Whiff.
Probably need to use dynamic objects.

**Comment by TheFireCookie** [comment]
It worked in NEST 7.x so I guess we could/should take inspiration from the old code to implement it in the new client.

**Comment by timschwallie** [comment]
From what I see, the real issue is the JSON produced from ES. Fix that and a lot of the client code on many platforms becomes much easier to code, maintain, and performs better. Basically a lot of band-aiding going on with client code.
I pretty sure another approach I have in mind would work with current JSON, result in easier to code to maintain and faster, but I need more coffee.
